### PR TITLE
migration: enable migrating from non-root to root

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
The migration code allows VMIs to migrate from root to non-root, but not the other way around.
Sounds like an oversight to me, this PR addresses that.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2091911

**Special notes for your reviewer**:
Will write unit/functional test(s) if preliminary reviews are positive.

**Release note**:

```release-note
Non-root VMs will now migrate to root VMs after a cluster disables non-root.
```

### CODE REVIEW

**Comment by stu-gott** [comment]
/approve

Other than the minor nit mentioned by acardace, this looks to be in good shape.

File: pkg/virt-controller/watch/migration.go [hunk 1]
```diff
@@ -1064,18 +1066,28 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
...
+				if vmi.Annotations != nil {
+					if _, ok := vmi.Annotations[virtv1.DeprecatedNonRootVMIAnnotation]; ok {
+						patches = append(patches, fmt.Sprintf(`{ "op": "remove", "path": "/metadata/annotations/%s"}`, controller.EscapeJSONPointer(virtv1.DeprecatedNonRootVMIAnnotation)))
 					}
 				}
+			}
+			if len(patches) != 0 {
+				vmi, err = c.clientset.VirtualMachineInstance(vmi.Namespace).Patch(vmi.Name, types.JSONPatchType, controller.GeneratePatchBytes(patches), &v1.PatchOptions{})
+				if err != nil {
+					return fmt.Errorf("failed to mark VMI as nonroot: %v", err)
```
[the remark this answers is not in the document]
Comment (jean-edouard) [reply pkg/virt-controller/watch/migration.go]: Done

**Review by xpivarc — APPROVED** [review]

File: tests/migration_test.go [hunk 2]
```diff
@@ -1741,12 +1743,98 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
...
+					tests.EnableFeatureGate(virtconfig.NonRoot)
+				}
+			})
+			AfterEach(func() {
+				if clusterIsRoot {
+					tests.DisableFeatureGate(virtconfig.NonRoot)
+				} else {
+					tests.EnableFeatureGate(virtconfig.NonRoot)
+				}
+				if dv != nil {
```
Comment (xpivarc) [inline tests/migration_test.go:1769]: This should not be needed. I will clean it up later.

**Pull request merged**